From 960714e849b62253f30fbe23df67e0b62fda1cd2 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@linux.spacemit.com>
Date: Wed, 21 May 2025 13:54:19 +0800
Subject: [PATCH 23/43] riscv: k1: dts: enable pwm for bananapi f3 board

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/26
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/6cd071472701900fb059c20e119fb25f187b41ea

--------------------------------

There is a pwm available on bananapi f3, and the pin is
loacted at the 26pin gpio interface.

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>
---
 arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts b/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts
index c96b4e5cd6f12..be4483d8f13cb 100644
--- a/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts
+++ b/arch/riscv/boot/dts/spacemit/k1-bananapi-f3.dts
@@ -114,3 +114,9 @@ &i2c8 {
 	pinctrl-0 = <&pinctrl_i2c8>;
 	status = "okay";
 };
+
+&pwm7 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm7_0>;
+	status = "okay";
+};
-- 
2.51.0

